src/integrationtests/java/com/aws/greengrass/integrationtests/ipc/VerifyClientDeviceIdentityTest.java

```diff
@@ -218,7 +216,7 @@ void GIVEN_broker_WHEN_verify_client_identity_with_validation_exception_THEN_err
     }
 
     @Test
-    void GIVEN_broker_WHEN_verify_client_identity_with_internal_server_exception_THEN_error_is_thrown(
+    void GIVEN_broker_WHEN_verify_client_identity_with_internal_server_exception_THEN_verify_with_cached_result(
```

Comment:
Hmm, this is a behavior change that we need think about carefully. Let's chat tomorrow about it

Reply:
I changed the offline-auth behavior to [throw exception when offline with uncached auth result](https://github.com/aws-greengrass/aws-greengrass-client-device-auth/pull/104/commits/e7323670266c8895834dddd905fef08b90a482bc), for both CertificateRegistry and ThingRegistry. Is this okay?